### PR TITLE
De-dupe search bar components

### PR DESCRIPTION
**Pull request by ajpal**
Sound Library:
Before:
image

After:
image

Icon Library:
Before:
image

After:
image

Animation Library:
Before:
image

After:
image

### CODE REVIEW

File: apps/src/templates/SearchBar.jsx [hunk 1]
```diff
@@ -31,32 +31,33 @@ const styles = {
...
+  searchArea: {
     position: 'relative'
   }
 };
 
 export default class AnimationPickerSearchBar extends React.Component {
   static propTypes = {
-    value: PropTypes.string.isRequired,
+    placeholderText: PropTypes.string.isRequired,
+    styles: PropTypes.object.isRequired,
```
Comment (joshlory) [inline apps/src/templates/SearchBar.jsx]: This feels a little weird but I can't 100% put my finger on why. Allowing the consumer to style different elements in the component makes sense, and so does combining all those properties into one object (I think). Maybe we should be explicit about what keys it takes:

```suggestion
    styles: PropTypes.object.shape({
      searchArea: PropTypes.object,
      icon: PropTypes.object,
      input: PropTypes.object
    });
```

What do you think?

Comment (ajpal) [reply apps/src/templates/SearchBar.jsx]: oo, didn't know you could specify what keys the object has! Yeah that definitely feels better, thanks!

Comment (joshlory) [reply apps/src/templates/SearchBar.jsx]: Do all of these styles _need_ to be customized? Or can we make smart defaults, and have them be the same everywhere `<SearchBar/>` gets used?

File: apps/src/templates/SearchBar.jsx [hunk 2]
```diff
@@ -31,32 +31,37 @@ const styles = {
     top: 9,
     left: 10
   },
-  search: {
+  searchArea: {
     position: 'relative'
   }
 };
 
 export default class AnimationPickerSearchBar extends React.Component {
```
Comment (joshlory) [inline apps/src/templates/SearchBar.jsx]: Nit: `SearchBar`?

File: apps/src/templates/SearchBar.jsx [hunk 3]
```diff
@@ -31,32 +31,37 @@ const styles = {
...
   static propTypes = {
-    value: PropTypes.string.isRequired,
+    placeholderText: PropTypes.string.isRequired,
+    styles: PropTypes.object.shape({
+      searchArea: PropTypes.object,
+      icon: PropTypes.object,
+      input: PropTypes.object
+    });
```
Comment (joshlory) [inline apps/src/templates/SearchBar.jsx]:
```suggestion
    }),
```

File: apps/src/templates/SearchBar.jsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  };
+
+  componentDidMount() {
+    this.searchBox.focus();
+  }
+
+  render() {
+    let customStyles = this.props.styles || {};
+    return (
+      <div style={{...styles.searchArea, ...customStyles.searchArea}}>
```
Comment (joshlory) [inline apps/src/templates/SearchBar.jsx]: Are these style overrides still needed? It doesn't look like any consumer is passing these props in.

**Review by joshlory — APPROVED** [review]
Nice cleanup!

File: apps/src/templates/SearchBar.story.jsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import SearchBar from './SearchBar';
+
+export default storybook => {
+  storybook.storiesOf('SearchBar', module).addStoryTable([
+    {
+      name: 'Search Bar',
+      story: () => (
+        <SearchBar
+          placeholderText={'Some Text'}
+          styles={{}}
```
Comment (joshlory) [inline apps/src/templates/SearchBar.story.jsx]: Nit: not used?

**Pull request merged**